coderd/workspacebuilds.go

```diff
@@ -874,13 +879,25 @@ func (api *API) workspaceBuildsData(ctx context.Context, workspaceBuilds []datab
 		return workspaceBuildsData{}, err
 	}
 
+	appIDs := make([]uuid.UUID, 0)
+	for _, app := range apps {
+		appIDs = append(appIDs, app.ID)
+	}
+
+	// nolint:gocritic // Getting workspace app statuses by app IDs is a system function.
+	statuses, err := api.Database.GetWorkspaceAppStatusesByAppIDs(dbauthz.AsSystemRestricted(ctx), appIDs)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return workspaceBuildsData{}, xerrors.Errorf("get workspace app statuses: %w", err)
+	}
```

Comment:
Why do we need the app statuses on the workspace builds as well? Won't we have them on the agent?

Reply:
This is just getting them on fetch. This actually does apply to the workspace agent anyways, just doing it in bulk to avoid separate queries.